### PR TITLE
Auto Generate Structural Schema for CRDs

### PR DESCRIPTION
**Pull request by jasonwzm**
We added functionality to autogenerate structural schema based on our protos in Cue generator. This PR is to generate the CRDs for the first time. The generated file is at kubernetes/customresourcedefinitions.gen.yaml. 

With the above change, the generated json files are also structural which still align with the openapi schema standard.

Also removed the cue image in the Makefile (now integrated with the build-tool image).

### CODE REVIEW

File: networking/v1alpha3/destination_rule.json [hunk 1]
```diff
@@ -245,8 +245,7 @@
           "LEAST_CONN",
           "RANDOM",
           "PASSTHROUGH"
-        ],
-        "default": "ROUND_ROBIN"
+        ]
```
Comment (ozevren) [inline networking/v1alpha3/destination_rule.json]: What is the impact of the removal of the default field?

**Comment by geeknoid** [comment]
@jasonwzm Do we still want to land this?

**Comment by jasonwzm** [comment]
@geeknoid yes, still working on it. discussed with Ed that we will add more support to the `description` field.

**Comment by jasonwzm** [comment]
Closing, in favor of https://github.com/istio/api/pull/1142

**Pull request closed without merging**